src/main/java/org/folio/circulation/domain/UpdateItem.java

```diff
@@ -44,9 +44,8 @@ private Item changeItemOnCheckIn(
 
     if (requestQueue.hasOutstandingFulfillableRequests()) {
       Request request = requestQueue.getHighestPriorityFulfillableRequest();
-      String pickupServicePointIdString = request.getPickupServicePointId();
-      UUID pickUpServicePointId = pickupServicePointIdString == null ? null : UUID.fromString(request.getPickupServicePointId());
-      if (pickUpServicePointId == null || checkInServicePointId == null || checkInServicePointId.equals(pickUpServicePointId)) {
+      UUID pickUpServicePointId = UUID.fromString(request.getPickupServicePointId());
```

Comment:
Hmm... This is a fair point. I believe a pick up service point won't be set if the request is set to delivery. I'll look into this...

Reply:
So I was unable to get the code to fail. I am not sure how delivery fulfillment works or if it is fully implemented. Perhaps I should just protect the `UUID.fromString()` so `null` won't throw an exception.

Reply:
A pickup service point *could* but *should not* be set for delivery.

This code should only be for hold shelf fulfilment. 

That said, I tend to like explicit null guards.

Delivery fulfilment is not implemented at all, the queue skips over any requests that are of that type. They only exist because they were added too eagerly, and removing them would be breaking.

Reply:
It appears requests consider delivery fulfillment not "fulfillable". Only hold shelf requests are considered fulfillable, so `requestQueue.getHighestPriorityFulfillableRequest();` should only return hold shelf requests which have to have a pick up service point set. So I think we are OK for now. At least this is my interpretation.